app/controllers/slack_controller.rb

```diff
@@ -3,11 +3,17 @@ class SlackController < ApplicationController
 
   CannotPlusOneYourself = Class.new(StandardError)
   MissingRecipient = Class.new(StandardError)
+  MissingSlackToken = Class.new(StandardError)
 
   def plus
     ActiveRecord::Base.transaction do
+      team = Team.find_or_initialize_by(slack_team_id: params[:team_id])
+      recipient_name = MessageParser.new(text_params[:text], text_params[:trigger_word]).recipient_name
+      raise MissingSlackToken if team.slack_token.nil? && recipient_name.start_with?("<@")
```

Comment:
it's not only when the token is missing but also when it's invalid (for any reason). that's actually the case in our organization now, that a token is invalid